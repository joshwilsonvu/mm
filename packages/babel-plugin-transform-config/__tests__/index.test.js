
const pluginTester = require('babel-plugin-tester').default;
const plugin = require('..');

// replaces project path with <PROJECT_ROOT> in absolute paths
const jestSerializerPath = require('jest-serializer-path');
expect.addSnapshotSerializer(jestSerializerPath);

pluginTester({
  plugin: plugin,
  tests: {
    "adds _component, _path to modules": {
      code: `
        export default {
          port: 8080,
          modules: [
            {
              module: "a-module",
              position: "top_right"
            }
          ]
        };
      `,
      snapshot: true,
    },
    // correctly resolves modules in a "modules/default" folder
    "resolves default modules": {
      code: `
        export default {
          modules: [
            {
              module: "a-default-module"
            }
          ]
        };
      `,
      snapshot: true,
    },
    "throws for nonexistent modules": {
      code: `
        export default {
          modules: [
            {
              module: "a-nonexistent-module"
            }
          ]
        };
      `,
      error: SyntaxError,
    },
    "doesn't search if option `modulesPath` is specified": {
      code: `
        export default {
          modules: [
            {
              module: "the-module"
            }
          ]
        };
      `,
      snapshot: true,
      pluginOptions: {
        modulesPath: "../../custom-modules",
      }
    }
  },
  babelOptions: {
    filename: require.resolve('./config/dummy'),
    babelrc: false,
    configFile: false,
    root: __dirname,
  },
});
