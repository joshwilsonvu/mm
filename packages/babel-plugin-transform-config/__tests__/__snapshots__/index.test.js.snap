// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-transform-config adds _component, _path to modules: adds _component, _path to modules 1`] = `

export default {
  port: 8080,
  modules: [
    {
      module: "a-module",
      position: "top_right"
    }
  ]
};

      ↓ ↓ ↓ ↓ ↓ ↓

export default {
  port: 8080,
  modules: [
    {
      module: "a-module",
      _component: require("react").lazy(() =>
        import(
          /* webpackChunkName: "a-module" */

          /* webpackPreload: true */
          "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/a-module/a-module.js"
        )
      ),
      _path:
        "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/a-module/a-module.js",
      position: "top_right",
    },
  ],
};


`;

exports[`babel-plugin-transform-config bails if .disabled === true: bails if .disabled === true 1`] = `

export default {
  modules: [
    {
      module: "a-module",
      disabled: true
    }
  ]
};

      ↓ ↓ ↓ ↓ ↓ ↓

export default {
  modules: [
    {
      module: "a-module",
      disabled: true,
    },
  ],
};


`;

exports[`babel-plugin-transform-config doesn't search if option \`modulesPath\` is specified: doesn't search if option \`modulesPath\` is specified 1`] = `

export default {
  modules: [
    {
      module: "the-module"
    }
  ]
};

      ↓ ↓ ↓ ↓ ↓ ↓

export default {
  modules: [
    {
      module: "the-module",
      _component: require("react").lazy(() =>
        import(
          /* webpackChunkName: "the-module" */

          /* webpackPreload: true */
          "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/custom-modules/the-module/index.js"
        )
      ),
      _path:
        "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/custom-modules/the-module/index.js",
    },
  ],
};


`;

exports[`babel-plugin-transform-config resolves default modules: resolves default modules 1`] = `

export default {
  modules: [
    {
      module: "a-default-module"
    }
  ]
};

      ↓ ↓ ↓ ↓ ↓ ↓

export default {
  modules: [
    {
      module: "a-default-module",
      _component: require("react").lazy(() =>
        import(
          /* webpackChunkName: "a-default-module" */

          /* webpackPreload: true */
          "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/default/a-default-module/a-default-module.js"
        )
      ),
      _path:
        "<PROJECT_ROOT>/packages/babel-plugin-transform-config/__tests__/modules/default/a-default-module/a-default-module.js",
    },
  ],
};


`;
